contrib/node/src/python/pants/contrib/node/tasks/node_run.py

```diff
@@ -32,11 +32,22 @@ def execute(self):
     if self.is_node_module(target):
       node_paths = self.context.products.get_data(NodePaths)
       node_path = node_paths.node_path(target)
-
-      args = ['run-script', self.get_options().script_name, '--'] + self.get_passthru_args()
-
-      with pushd(node_path):
-        result, npm_run = self.execute_npm(args, workunit_labels=[WorkUnitLabel.RUN])
-        if result != 0:
-          raise TaskError('npm run script failed:\n'
-                          '\t{} failed with exit code {}'.format(npm_run, result))
+      package_manager = self.get_package_manager_for_target(target=target)
+      if package_manager == 'npm':
```

Comment:
It seems like an overkill to extract an abstract class for a simple string comparison. I will extract a const list to reduce the concern of error prone.